dist/systemd/system/greenboot-trigger.service

```diff
@@ -1,21 +0,0 @@
-[Unit]
-Description=Greenboot - TODO 2
-DefaultDependencies=no
-Conflicts=shutdown.target
-Before=shutdown.target
-
-Wants=local-fs.target
-After=local-fs.target
-
-Before=multi-user.target systemd-update-done.service
-ConditionNeedsUpdate=|/etc
-ConditionNeedsUpdate=|/var
-
-[Service]
-Type=oneshot
-RemainAfterExit=true
-ExecStart=/usr/libexec/greenboot/greenboot stamp
-Restart=no
-
-[Install]
-WantedBy=multi-user.target
```

Comment:
I’m wondering why this got removed altogether? And the other service got a WantedBy? This change is mainly around boot-complete and the way it should work so I want to make sure we know exactly how this process work. ConditionNeedsUpdate was a key piece of all this but it’s been discarded, can you clarify? I don’t want us to just rewrite greenboot in rust, it has to gain robustness and the problem was the wiring with systemd not the language per se

Reply:
also, without this, how do I test that I need to do a check on reboot/upgrade? and how does it not just reboot loop?

Reply:
so for testing purpose as long as grub handles auto decrement of boot_counter on every restart, it will stop once it reaches -1.

Reply:
probably it can let me check that in a bit.

Reply:
have included this in the PR, renamed it to greenboot-rollback.service as it will check when boot_counter=-1 .